### PR TITLE
[gpio] Added parameter for pull up/down resistor and other improvements

### PR DESCRIPTION
**Pull request by SloCompTech**
*[gpio] Added parameter for pull up/down resistor for GPIO binding*

Changes:

- Added Pull Up/Down parameter to input GPIO pin configuration
- GPIO pin mode set with library constant instead of integer
- Updated README.md of binding (added information about new parameter)
- Improved README.md of binding with some additional info
- Also ran static code analysis
- Tested on my setup, no problems detected

Most of changes were implemeted by @sjoerdtakken ([Forum post](https://community.openhab.org/t/gpio-new-gpio-binding-for-oh3/113943/33), [His repo with changes](https://wordpressvonmorgen.dd-dns.de/pigpio/)), but not commited to this repo. 

Here is compiled version: [org.openhab.binding.gpio-3.1.0.zip](https://github.com/openhab/openhab-addons/files/6566972/org.openhab.binding.gpio-3.1.0.zip)

### CODE REVIEW

**Comment by SloCompTech** [comment]
With little "workout" fixed issues with signing-off commits

File: bundles/org.openhab.binding.gpio/README.md [hunk 1]
```diff
@@ -34,14 +34,16 @@ sudo nano /etc/systemd/system/pigpiod.service.d/public.conf
...
-Now that Remote GPIO is enabled, get the daemon going:
+Now that Remote GPIO is enabled, get the daemon going (even if installed with apt-get):
 ```
 sudo systemctl enable pigpiod 
 sudo systemctl start pigpiod
 ```
 
 In openHAB, set `host` to the address of the pi and the `port` to the port of pigpio (default: 8888).
 
+Note: If you are running Pigpio on same host as OpenHab, then set host to **::1**.
```
Comment (martinvw) [inline bundles/org.openhab.binding.gpio/README.md:45]: Please spell as openHAB

File: bundles/org.openhab.binding.gpio/src/main/resources/OH-INF/thing/pigpio-remote.xml [hunk 2]
```diff
@@ -50,6 +50,17 @@
 				<default>10</default>
 				<advanced>true</advanced>
 			</parameter>
+			<parameter name="pullupdown" type="integer" min="0" max="2">
+				<label>Pull Up/Down resistor</label>
```
Comment (martinvw) [inline bundles/org.openhab.binding.gpio/src/main/resources/OH-INF/thing/pigpio-remote.xml:54]:
```suggestion
				<label>Pull Up/Down Resistor</label>
```

File: bundles/org.openhab.binding.gpio/README.md [hunk 3]
```diff
@@ -78,7 +81,7 @@ Thing gpio:pigpio-remote:sample-pi-1 "Sample-Pi 1" [host="192.168.2.36", port=88
 Thing gpio:pigpio-remote:sample-pi-2 "Sample-Pi 2" [host="192.168.2.37", port=8888] {
     Channels:
         Type pigpio-digital-input : sample-input-3 [ gpioId=16, debouncingTime=20]
-        Type pigpio-digital-input : sample-input-4 [ gpioId=17, invert=true, debouncingTime=5]
+        Type pigpio-digital-input : sample-input-4 [ gpioId=17, invert=true, debouncingTime=5, pullupdown=2]
```
Comment (martinvw) [inline bundles/org.openhab.binding.gpio/README.md:84]: It feels like a very direct way, would it make sense to expose a bit less direct, or not. I'm in doubt.

File: bundles/org.openhab.binding.gpio/README.md (hunk 1 above)
Comment (SloCompTech) [reply bundles/org.openhab.binding.gpio/README.md]: Fixed spelling

File: bundles/org.openhab.binding.gpio/src/main/resources/OH-INF/thing/pigpio-remote.xml (hunk 2 above)
Comment (SloCompTech) [reply bundles/org.openhab.binding.gpio/src/main/resources/OH-INF/thing/pigpio-remote.xml]: Fixed spelling

File: bundles/org.openhab.binding.gpio/README.md (hunk 3 above)
Comment (SloCompTech) [reply bundles/org.openhab.binding.gpio/README.md:84]: Have you got any suggestion for improvement ?

**Comment by martinvw** [comment]
@fwolter what do you think?

File: bundles/org.openhab.binding.gpio/README.md (hunk 3 above)
Comment (fwolter) [reply bundles/org.openhab.binding.gpio/README.md:84]: Do you mean like `pullupdown=OFF`, `pullupdown=UP`, `pullupdown=DOWN`? Would be better from the usability perspective, IMHO. But I won't insist on changing it.

Comment (martinvw) [reply bundles/org.openhab.binding.gpio/README.md:84]: Let's merge it then

**Review by martinvw — APPROVED** [review]

File: bundles/org.openhab.binding.gpio/README.md (hunk 3 above)
Comment (fwolter) [reply bundles/org.openhab.binding.gpio/README.md:84]: OK, @SloCompTech feel free to file a follow-up PR if you agree on these thoughts.

Comment (SloCompTech) [reply bundles/org.openhab.binding.gpio/README.md:84]: Ok, will do, so I need to change parameter type to string or is there another more suitable type like a constant ?

Comment (fwolter) [reply bundles/org.openhab.binding.gpio/README.md:84]: There's no enum type or so. You need to make it `text`. See https://www.openhab.org/docs/developer/bindings/config-xml.html#xml-structure-for-configuration-descriptions for the available types.

**Comment by SloCompTech** [comment]
I created follow up PR #10862, but I have problems with signing-off, since this merge request wasn't signed, new PR doesn't pass DCO. Help would be appreciated to fix this, since this could affect other PRs.

**Pull request merged**